test/testStaroddi.m

```diff
@@ -0,0 +1,249 @@
+
+            varid = getVar(new_data.variables, 'PSAL_2');
+            var = new_data.variables{varid};
+            assert(contains(var.comment, 'mbar was adjusted to pressure'));
+        end
+
+        function testStaroddiParser_mini_v000(~, starmon_mini_v000_param, mode)
+            new_data = StaroddiParser(starmon_mini_v000_param, mode);
+            assert(strcmp(new_data.meta.instrument_make, 'Star ODDI'));
+            assert(strcmp(new_data.meta.instrument_model, 'Starmon Mini'));
+            assert(length(new_data.dimensions) == 1);
+            assert(length(new_data.variables) > 4);
+        end
+
+        function testStaroddiParser_mini_v001(~, starmon_mini_v001_param, mode)
+            new_data = StaroddiParser(starmon_mini_v001_param, mode);
+            assert(strcmp(new_data.meta.instrument_make, 'Star ODDI'));
+            assert(strcmp(new_data.meta.instrument_model, 'Starmon mini')); % yes - mini lowercase now...
+            assert(length(new_data.dimensions) == 1);
+            assert(length(new_data.variables) > 4);
+        end
+
+        function testStaroddiParser_dsttilt_v000(~, starmon_dsttilt_v000_param, mode)
+            new_data = StaroddiParser(starmon_dsttilt_v000_param, mode);
+            assert(strcmp(new_data.meta.instrument_make, 'Star ODDI'));
+            assert(contains(new_data.meta.instrument_model, 'DST Tilt'));
+            assert(length(new_data.dimensions) == 1);
+            assert(length(new_data.variables) > 4);
+        end
+
+        function testStaroddiParser_dstctd_v000(~, starmon_dstctd_v000_param, mode)
+            new_data = StaroddiParser(starmon_dstctd_v000_param, mode);
+            assert(strcmp(new_data.meta.instrument_make, 'Star ODDI'));
+            assert(contains(new_data.meta.instrument_model, 'DST CTD'));
+            assert(length(new_data.dimensions) == 1);
+            assert(length(new_data.variables) > 4);
+        end
+
+    end
+
+end
+
+function [m, d, v] = loadfields(idata)
+%
+% grab metadata, dimensons and variables
+% A one-liner to save 4 lines.
+%
+m = rmfield(idata.meta, 'header');
+m = orderfields(m); % required for tree matching.
+d = idata.dimensions;
+v = idata.variables;
+end
+
+function [param, files, folder] = get_testfiles(maker, model, version)
+%
+% get param,files, and folder testfiles information for a certain
+% instrument maker/model and version.
+%
+root_folder = toolboxRootPath();
+folder = fullfile(root_folder, 'data', 'testfiles', maker, model, version);
+files = FilesInFolder(folder);
+param = files2namestruct(files);
+end
+
+function [folder] = get_folder(maker, model, version)
+%
+% wrapper to obtain only folder
+%
+[~, ~, folder] = get_testfiles(maker, model, version);
+end
+
+function save_mat(data, tparam, outfolder)
+%
+% save a mat file to outfolder/tparam.mat
+%
+fs = char(tparam);
+file = [basename(fs) '.mat'];
+omat = fullfile(outfolder, file);
+save(omat, 'data');
+end
```

Comment:
I can see there are tests for the new staroddi parser, but it feels like it would be great to have specific unittests for the generic parser functions, as well as the new utils functions you have implemented. Most of the tests are already written in the series of example you wrote

Reply:
Yeap - I will add them